src/build_system.ml

```diff
@@ -89,6 +89,11 @@ module Internal_rule = struct
 
   let loc ~file_tree ~dir t  = rule_loc ~file_tree ~dir ~loc:t.loc
 
+  let decode : t Dune_lang.Decoder.t =
+    let open Dune_lang.Decoder in
+    loc >>= fun loc ->
+    Errors.fail loc "<not-implemented>"
```

Comment:
If this is failing decoder is specific to memoization I think it's be better to move it to the `Memo.Make` argument.

Reply:
Indeed, I reworked a bit the API

Reply:
thanks!